### ISSUE TITLE
Implement type hierarchy 

### ISSUE THREAD

**Issue by themkat** [fwcd/kotlin-language-server#348]
In LSP 3.17.0, [type hierarchy requests](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_prepareTypeHierarchy) (textDocument/typeHierarchy) was introduced. Can be super useful when you have more than one super type, or subtype, and want to see how everything fit together. Sometimes we have complex inheritance structures, especially when using Java code...

Might look into this myself if I find the time, but making an issue so we don't forget about it.

**Comment by fwcd** [comment]
Yes, type hierarchy would be great, maybe we could implement this by capturing the subtype relationships in the index?
